templates/ingress.yaml

```diff
@@ -18,7 +18,13 @@
   {{- end }}
 {{- end }}
 {{- end }}
+{{- if semverCompare ">=1.19-0" $.Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" $.Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
 apiVersion: extensions/v1beta1
+{{- end }}
```

Comment:
I don't think we want to actually check the k8s version here. Instead please use `Capabilities.APIVersions.Has` which is easier to understand and maintain for these checks. https://helm.sh/docs/chart_template_guide/builtin_objects/